### PR TITLE
fix: prevent blank KG chat replies when extract-relationships runs last

### PR DESCRIPTION
**Pull request by josephfung**
## Summary

- **Root cause**: The coordinator system prompt instructs the agent to call `extract-relationships` after every message. When it ran as the *last* tool call (e.g. after `query-relationships`), the LLM would produce `stop_reason=end_turn` with an empty content array, delivering a `{"reply":""}` to the user.

- **Prompt fix** (`agents/coordinator.yaml`): Clarifies that `extract-relationships` is a silent background housekeeping task that must not replace the user-facing text response. Instructs the agent to always write its reply first.

- **Runtime fix** (`src/agents/runtime.ts`): Defensive fallback — detects `content === ''` after tool use, logs an `error`, and returns a safe message instead of forwarding the empty string.

- **Provider fix** (`src/agents/llm/anthropic.ts`): Elevates the empty-text log from `warn` to `error` so the `stop_reason` is visible at the right severity when correlated in production logs.

- **Regression test** (`tests/unit/agents/runtime.test.ts`): Simulates `tool_use → empty text` sequence; asserts fallback is delivered.

## Test plan

- [ ] All unit tests pass (`npx vitest run tests/unit/agents/runtime.test.ts`)
- [ ] Manually ask the KG chat "What relationships do you know about X?" and verify a non-empty reply is returned
- [ ] Check server logs confirm `extract-relationships` no longer causes blank replies

___

## **CodeAnt-AI Description**
**Prevent blank KG chat replies after relationship extraction runs**

### What Changed
- KG chat now returns a safe fallback message instead of an empty reply when the assistant finishes a turn without any user-facing text after running `extract-relationships`.
- The coordinator instructions now make it clear that relationship extraction is a background step and must not replace the reply shown to the user.
- Empty text responses after tool use are now treated as errors in logs, and a regression test covers the blank-reply case.

### Impact
`✅ Fewer blank KG chat replies`
`✅ Clearer KG chat failures`
`✅ More reliable relationship extraction handling`
<details>
<summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

File: src/agents/runtime.ts [hunk 1]
```diff
@@ -407,7 +407,19 @@ export class AgentRuntime {
         { agentId, inputTokens: response.usage.inputTokens, outputTokens: response.usage.outputTokens },
         'Agent task completed',
       );
-      responseContent = response.content;
+      if (response.content === '') {
```
[the remark this answers is not in the document]
Comment (josephfung) [reply src/agents/runtime.ts]: Good catch —  would have bypassed the  check and delivered a visually blank reply. Fixed by changing the guard to . Also extended the regression test to cover the whitespace case alongside the empty-string case. Pushed in 8af7ae3.
